qmi/core/transport.py

```diff
@@ -179,16 +182,30 @@ def discard_read(self) -> None:
 
 
 class TransportDescriptorParser:
-
+    """This class is for creating a transport-specific parser classes and has (static) methods that are
+    used for parsing transport strings.
+    """
     def __init__(self,
                  interface: str,
                  positionals: List[Tuple[str, Tuple[Type, bool]]],
-                 keywords: Mapping[str, Tuple[Type, bool]]):
+                 keywords: MutableMapping[str, Tuple[Type, bool]]
```

Comment:
i missed this one, but if we want to get rid of `MutableMapping` this is the last place to change it. Then we don't need to import it either

Reply:
From https://docs.python.org/3/library/typing.html#typing.Dict, (`Mutable`)`Mapping` is preferred for input arguments, Thus I would like change this back to the original `Mapping`, as we do not need to mutate this input. The same goes for the `default_parameters` input in `parse_parameter_strings`.

Reply:
The changes for output types to `Dict` we can thus keep.